tests/sqlparser_common.rs

```diff
@@ -3446,32 +3446,45 @@ fn parse_joins_using() {
 
 #[test]
 fn parse_natural_join() {
-    fn natural_join(f: impl Fn(JoinConstraint) -> JoinOperator) -> Join {
+    fn natural_join(f: impl Fn(JoinConstraint) -> JoinOperator, alias: Option<TableAlias>) -> Join {
         Join {
             relation: TableFactor::Table {
                 name: ObjectName(vec![Ident::new("t2")]),
-                alias: None,
+                alias,
                 args: None,
                 with_hints: vec![],
             },
             join_operator: f(JoinConstraint::Natural),
         }
     }
+
+    // if not specified, inner join as default
     assert_eq!(
         only(&verified_only_select("SELECT * FROM t1 NATURAL JOIN t2").from).joins,
-        vec![natural_join(JoinOperator::Inner)]
+        vec![natural_join(JoinOperator::Inner, None)]
     );
+    // left join explicitly
     assert_eq!(
         only(&verified_only_select("SELECT * FROM t1 NATURAL LEFT JOIN t2").from).joins,
-        vec![natural_join(JoinOperator::LeftOuter)]
+        vec![natural_join(JoinOperator::LeftOuter, None)]
     );
+
+    // right join explicitly
     assert_eq!(
         only(&verified_only_select("SELECT * FROM t1 NATURAL RIGHT JOIN t2").from).joins,
-        vec![natural_join(JoinOperator::RightOuter)]
+        vec![natural_join(JoinOperator::RightOuter, None)]
     );
+
+    // full join explicitly
     assert_eq!(
         only(&verified_only_select("SELECT * FROM t1 NATURAL FULL JOIN t2").from).joins,
-        vec![natural_join(JoinOperator::FullOuter)]
+        vec![natural_join(JoinOperator::FullOuter, None)]
+    );
+
+    // natural join with alias
+    assert_eq!(
+        only(&verified_only_select("SELECT * FROM t1 NATURAL JOIN t2 AS t3").from).joins,
```

Comment:
this is an interesting one -- it adds the alias to the entire `NaturalJoin` rather than aliasing `t2 to t3` on one side of the input. 

Is that what other implementations do?

Reply:
postgresql has this

```bash
willy=# select * from test1 natural join test2 as test3;
 c1 | c0 | c2 | c3 
----+----+----+----
(0 rows)

willy=# select test3.c1 from test1 natural join test2 as test3;
 c1 
----
(0 rows)
```

Reply:
it alias `t2 to t3` but not alias to the entire `NaturalJoin`.

Reply:
perhaps i should delete this test? i thought at we need to protect the case that it incorrectly alias the whole natural join to t3

Reply:
I think it is a good test and that we should leave it in